lib/wp/api/client.rb

```diff
@@ -8,52 +8,66 @@ module WP::API
   class Client
     include HTTParty
     include Endpoints
+    attr_reader :args
 
     default_options.update(verify: false)
 
     attr_accessor :host
 
-    DIRECT_PARAMS = %w(type context filter)
+    DIRECT_PARAMS = %w[type context filter].freeze
 
-    def initialize(host:, scheme: 'http')
-      @scheme = scheme
-      @host = host
+    def initialize(args = {})
+      @scheme = args.dig(:scheme)
+      @host = args.dig(:host)
       @basic_auth = {}
       @proxy = {}
       @oauth = nil
-
       fail ':host is required' unless host.is_a?(String) && host.length > 0
     end
 
     def inspect
       to_s.sub(/>$/, '') + " @scheme=\"#{@scheme}\" @host=\"#{@host}\" @basic_auth=\"#{!@basic_auth.empty?}\" @oauth=\"#{!@oauth.nil?}\">"
     end
 
-    def basic_auth(username:, password:)
-      @basic_auth = {username: username, password: password}
+    def basic_auth
```

Comment:
It's wrong, in this case, you can't assign dynamically, only for one resource from configuration